Introducing JSX translation (#21)
reactjs/hu.reactjs.org: content/docs/introducing-jsx.md

---
id: introducing-jsx
title: JSX bemutatása
permalink: docs/introducing-jsx.html
prev: hello-world.html
next: rendering-elements.html
---

Vizsgáljuk meg a következő változó deklarálást:

```js
const element = <h1>Helló, világ!</h1>;
```

Ez a furcsa szintaktika se nem szöveg se nem HTML.

JSX-nek hívják, ami egy szintaxis kiterjesztés a JavaScript nyelvhez. Használatát ajánljuk a React-el, hogy leírd miként fog kinézni a felhasználói felület. A JSX egy sablonnyelvre emlékeztethet, de a JavaScript teljes erejével bír.

A JSX React "elemeket" állít elő. A [következő szekcióban](/docs/rendering-elements.html) fogjuk megvizsgálni a DOM-ba való renderelésüket. Lent megtalálod a JSX alapjait, amelyek az elinduláshoz szükségesek.

### Miért a JSX? {#why-jsx}

> A React felöleli a tényt, hogy a megjelenítési logika eleve összekapcsolódik a többi felhasználói felület logikájával: hogyan kezelik az eseményeket, hogyan változnak az állapotok az idő múlásával, és hogyan állnak elő az adatokat a megjelenítéshez.

Reviewer: ```suggestion
A React azt az elvet vallja, hogy a megjelenítési logika eleve összekapcsolódik a többi felhasználói felületi logikával: az események kezelésével, állapotok időbeni változásával, és hogy az adat hogyan van előkészítve annak megjelenítéséhez.
```

I am not sure about this one though.

> Ahelyett, hogy a *technológiákat* mesterségesen elválasztaná azzal, hogy a jelölést és a logikát külön fájlokba helyezi, a React szempontok szerint szeparál lazán összekapcsolt „komponenseknek” nevezett egységekkel, amelyek mindkettőt tartalmazzák. Vissza fogunk térni a komponensekhez egy [későbbi szekcióban](/docs/components-and-props.html), ha még nem vagy otthon JS-be a jelölés használatával, [ez a beszélgetés](https://www.youtube.com/watch?v=x7cQ3mrcKaY) meggyőzhet téged.

Reviewer: ```suggestion
Ahelyett, hogy a *technológiákat* mesterségesen elválasztaná azzal, hogy a jelölést és a logikát külön fájlokba helyezi, a React szempontok szerint szeparál lazán összekapcsolt „komponenseknek” nevezett egységekkel, amelyek mindkettőt tartalmazzák. Vissza fogunk térni a komponensekhez egy [későbbi szekcióban](/docs/components-and-props.html), ha még nem vagy otthon a jelölés JS-ben való használatával, [ez a beszélgetés](https://www.youtube.com/watch?v=x7cQ3mrcKaY) meggyőzhet téged.
```

Reviewer: @gergely-nagy forgot this one :)

A React [nem követeli meg](/docs/react-without-jsx.html) a JSX használatát, de a legtöbb ember hasznosnak találja mint vizuális segédeszköz ha felhasználói felülettel dolgozik JavaScript kódban. Ezenkívül azt is lehetővé teszi, hogy a React hasznosabb hiba- és figyelmeztető üzeneteket jelenítsen meg.

Ezeket tisztázva, kezdjünk is bele!

### Beágyazott kifejezések JSX-ben {#embedding-expressions-in-jsx}

> A következő példában egy `name` nevű változót deklarálunk, majd a JSX-ben használjuk úgy, hogy kapcsos zárójelek közé csomagoljuk:

Reviewer: ```suggestion
A következő példában egy `name` változót deklarálunk, majd a JSX-ben használjuk úgy, hogy kapcsos zárójelbe tesszük:
```

```js{1,2}
const name = 'Josh Perez';
const element = <h1>Helló, {name}</h1>;

ReactDOM.render(
  element,
  document.getElementById('root')
);
```

> A JSX-be bármilyen [JavaScript kifejezést](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_Operators#Expressions) beilleszthetsz kapcsos zárójelek közé. Például a `2 + 2`, `user.firstName`, vagy a `formatName(user)` mind érvényes JavaScript kifejezések.

Reviewer: ```suggestion
A JSX-be bármilyen érvényes [JavaScript kifejezést](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_Operators#Expressions) beilleszthetsz kapcsos zárójelek közé. Például a `2 + 2`, `user.firstName`, vagy a `formatName(user)` mind érvényes JavaScript kifejezések.
```

[](codepen://introducing-jsx)

> A JSX-et több sorra bontottuk az olvashatóság érdekében. Bár ez nem kötelező, de ha ezt csinálod akkor javasoljuk, hogy csomagold zárójelek közé, elkerülve az [automatikus pontosvessző beillesztésének](https://stackoverflow.com/q/2846283) buktatóit.

Reviewer: ```suggestion
A JSX-et több sorra bontottuk az olvashatóság érdekében. Bár ez nem kötelező, de ha így csinálod akkor azt javasoljuk, hogy tedd zárójelek közé, hogy elkerüld az [automatikus pontosvessző beillesztésének](https://stackoverflow.com/q/2846283) buktatóit.
```

```js
const element = <img src={user.avatarUrl}></img>;
```

Ne használj idézőjeleket kapcsos zárójelek körül, mikor JavaScript kifejezést ágyazol be egy attribútumba. Használhatsz idézőjeleket (karakterláncokhoz) vagy kapcsos zárójelet (kifejezésekhez), de mindkettőt egy attribútumon belül ne.

>**Figyelem:**
>
>Mivel a JSX közelebb áll a JavaScripthez, mint a HTML-hez, a REACT DOM a `camelCase` elnevezési stílust használja a HTML attribútumnevek helyett.
>
>Például a `class` JSX-ben [`className`](https://developer.mozilla.org/en-US/docs/Web/API/Element/className)-re változik, a `tabindex`-et pedig [`tabIndex`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/tabIndex)-ként tudjuk használni.

### Gyermekek megadása JSX-ben {#specifying-children-with-jsx}

Ha egy címke üres, akkor azonnal lezárhatod egy `/>` (önlezáró) taggel, mint ahogy XML-ben is:

```js
const element = <img src={user.avatarUrl} />;
```

A JSX-címkék gyermekeket is tartalmazhatnak:

```js
const element = (
  <div>
    <h1>Helló!</h1>
    <h2>Jó itt látni.</h2>
  </div>
);
```

### A JSX megakadályozza az injekciós támadásokat {#jsx-prevents-injection-attacks}

> Biztonságos a felhasználói bevitel beágyazása a JSX-be:

Reviewer: ```suggestion
A felhasználói bevitel beágyazása a JSX-be biztonságos:
```

```js
const title = response.potentiallyMaliciousInput;
// Ez biztonságos:
const element = <h1>{title}</h1>;
```

> Alapértelmezés szerint a React DOM [elkerüli](https://stackoverflow.com/questions/7381974/which-characters-need-to-be-escaped-on-html) az összes JSX-be beágyazott értéket, mielőtt azokat megjelenítené. Így tudja biztosítani, hogy soha nem tudsz bármi olyat injektálni, amit kifejezetten nem a te alkalmazásodban írtak. Minden átalakul karakterlánccá, mielőtt megjelenítésre kerül. Ez segít megelőzni az [XSS (cross-site-scripting)](https://en.wikipedia.org/wiki/Cross-site_scripting) típusú támadásokat.

Reviewer: ```suggestion
Alapértelmezés szerint a React DOM [szanitálja](https://stackoverflow.com/questions/7381974/which-characters-need-to-be-escaped-on-html) az összes JSX-be beágyazott értéket, mielőtt azokat megjelenítené. Így biztosítja azt, hogy soha nem tudsz bármi olyat injektálni, amit kifejezetten nem a te alkalmazásodban írtak. Minden átalakul karakterlánccá, mielőtt megjelenítésre kerül. Ez segít megelőzni az [XSS (cross-site-scripting)](https://en.wikipedia.org/wiki/Cross-site_scripting) típusú támadásokat.
```
